storage/innobase/handler/ha_innodb.cc

```diff
@@ -15959,10 +15959,6 @@ ha_innobase::extra(
 	case HA_EXTRA_END_ALTER_COPY:
 		trx = check_trx_exists(ha_thd());
 		m_prebuilt->table->skip_alter_undo = 0;
-		if (!m_prebuilt->table->is_temporary()
-		    && !high_level_read_only) {
-			log_buffer_flush_to_disk();
-		}
```

Comment:
@vuvova As you can see, the removed code did avoid the unnecessary persistent log write when inserting into a temporary table.

Reply:
that's not what I mean. inserting into `LegArg` directly from `Leg` was slow, inserting into `LegArg` from a temporary table `tLegArg` was fast. This is what the reporter claims. I don't understand why `LegArg` cares where the row is coming from and I don't see how your fix explains that.

Reply:
@vuvova Did you look at the test scripts? The temporary table variant does have one `INSERT INTO non_temporary_table SELECT * FROM temporary_table` at the end, but the bulk of it consists of a large number of `INSERT…SELECT` statements that insert into a temporary table. Without this fix, there would be only one unnecessary log write, for the last `INSERT…SELECT` statement.

Reply:
No, I haven't, I only looked at the comment, my mistake. But if the temp table test script is doing that, then your change covers it and explains why it was fast. Thanks.